lib/DidKeyDriver.js

```diff
@@ -0,0 +1,202 @@
+/*!
+ * Copyright (c) 2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import {
+  Ed25519VerificationKey2020
+} from '@digitalbazaar/ed25519-verification-key-2020';
+import {
+  X25519KeyAgreementKey2020
+} from '@digitalbazaar/x25519-key-agreement-key-2020';
+
+import didContext from 'did-context';
+import {findVerificationMethod} from '@digitalbazaar/did-io';
+import ed25519Context from 'ed25519-signature-2020-context';
+import x25519Context from 'x25519-key-agreement-2020-context';
+
+export class DidKeyDriver {
+  /**
+   * @param {object} options - Options hashmap.
+   * @param {object} [options.verificationSuite=Ed25519VerificationKey2020] -
+   *   Key suite for the signature verification key suite to use.
+   * @param {Function} [options.convert] - An async function to convert a
+   *   verification key pair (like ed25519) to a key-agreement key pair (such
+   *   as x25519).
+   */
+  constructor({
+    verificationSuite = Ed25519VerificationKey2020,
+    convert = async verificationKeyPair => X25519KeyAgreementKey2020
```

Comment:
`convert` is not a very descriptive name ... `toKeyAgreementKeyPair` would be better -- or something more terse but less generic than `convert`. It's also not clear to me why we need the ability to set this hook at all -- why aren't converters built into the driver? And can devs get burned if they pass in a `verificationSuite` but don't pass in `convert` and it doesn't match?

Is the goal to make it such that when you're using the did:key driver you only have to support some subset of suites?

I'm not going to hold this up given timelines, but it feels like there's a more solid design we could come up with.

Reply:
@dlongley sure, happy to rename the property.
The idea with this design was to strike a balance between: make it easy and self-contained to use the latest suites, but also to make it flexible for other use cases (such as converting between an ed25519 2018 key and the 2020 x key, etc). Given that we're balancing at least 4 different suites right now, and that's before we even support other key types, there's no good way to build conversion into the driver.

Reply:
Then why pass the conversion function into the driver? Why not just have a different driver for every key type you want -- install the ones you support? They can each include a derived class from the basic driver if there's common functionality.

Reply:
I was more worried about the combinations of different _versions_ of key types (by year), but you know, since you mention it, I'm a proponent of YAGNI, so, I'm'a remove that parameter.

Reply:
@dlongley - removed the param.